pennylane/spin/lattice.py

```diff
@@ -190,7 +200,61 @@ def _generate_grid(self, neighbour_order):
             lattice_points.append(point)
             lattice_map.append(node_index)
 
-        return math.array(lattice_points), math.array(lattice_map)
+        return math.array(lattice_points), lattice_map
+
+    def get_custom_edges(self, custom_edges, lattice_map):
+        """Generates the edges described in `custom_edges` for all unit cells."""
+
+        if not all([len(edge) in (1, 2) for edge in custom_edges]):
+            raise TypeError(
+                """
+                custom_edges must be a list of tuples of length 1 or 2.
```

Comment:
```suggestion
                custom_edges must be a list of tuples, where the length of the tuple should be 1 or 2.
```

Reply:
Or something? From the latter functions, it seems that custom edges should look like this `[[(index, index), (op, coeff)], [(index, index), (op, coeff)]]` The description should be clearer on what custom edges should be.

Reply:
I would say custom edges is a list of list of tuples, where each inner list is of length 1 or 2, and contains tuples.

Reply:
The custom_edges can be a list of tuples of length 1 or 2, where you can just provide the edge in first tuple and the second tuple contains information about op and coeff, so something like this: [[(index, index), (op, coeff)], [(index, index), (op, coeff)]], where it is optional to provide the tuple containing (op, coeff), it could be [[(index, index)], [(index, index)]] as well. I will add an example nevertheless to make it clear.

Reply:
Ahh I see, but isn't that a list of list of tuples, not a list of tuples? 

**spaces added for clarity**
`custom_edges = [ [(index, index)], [(index, index)]   ]`

outer list has variable len
inner lists have len 1 or 2, where elements are tuples.

Reply:
yes, that is correct.

Reply:
Nevermind, I see the docstring now. Yep all good.